src/widgets/helper/ChannelView.cpp

```diff
@@ -1860,128 +1854,163 @@ void ChannelView::addContextMenuItems(
 
     auto menu = new QMenu;
     previousMenu = menu;
-
-    // Badge actions
-    if (creatorFlags.hasAny({MessageElementFlag::Badges}))
+    if (hoveredElement != nullptr)
```

Comment:
Wrapping this huge chunk in an `if/else` statement reduces readability of an already large method. I understand why you've done it, looks painful. Someone else might have a better idea on how to clean this up.

Reply:
We could instead handle the logic in a `if (hoveredElement == nullptr)`, but then the menu action order would be different than how it normally is (copy message before all other actions).

IIRC there is goto labels in C? Could probably make use of that in order to handle the logic as `==` and retain the menu action ordering, but I'm not sure if it's the best way to go about it.